### ISSUE TITLE
[Show Shift Edit] Visually indicate Shift status on changes

### ISSUE THREAD

**Issue by majick777** [netmix/radio-station#162]
Arising from #70 

When a new shift is added, that Shift could indicate "New" in the shift entry box, so that they can see where are editing. Considering if a user duplicates a shift at the top of the list, it will go to the bottom, but if they duplicate a few shifts, they won't know at which point the new ones started. Alternatively we could just have a heading separator for New Shifts when added.

Further, when an existing shift time is field is changed, it could indicate that the Shift is "Unsaved" in the shift entry box. So that if a user changes multiple entries or gets distracted before saving, they know what has been changed and are reminded visually they need to save their shift changes.

**Comment by tonyzeoli** [comment]
@majick777 Should we make all current saved show shifts have a light green background and any duplicate shifts have a yellow background until fully saved? So, for example, if you duplicate a shift, it turns yellow until it is updated and saved?

**Comment by tonyzeoli** [comment]
If an existing shift time is changed by the user, I think they would have to save it, no? If the shift time is changed, then maybe the row should immediately change color to yellow until saved and then revert back to green on save? Is there some interim Draft status, so if the user navigates away from the admin then yellow persists until they return and save?

**Comment by tonyzeoli** [comment]
Moved to 2.3.3

**Comment by majick777** [comment]
@tonyzeoli Yes nice one! I like the green and yellow background idea, this would certainly make it clear enough what is and has yet to be saved. 

I'm not sure what you mean about the interim Draft status, changes only exist on the page, if they navigate away without saving those changes will be lost. Autosaves ignore post meta like this because then you'd need to add a whole new interface asking which is the new and which is the old data (Personally I totally think WordPress could have had this a long time ago, but not sure if this was thought of or just put it in the too hard basket.) 

So I think it's overkill for us to create something like that just for the plugin too. However we can easily trigger the standard warning message "your changes have not been saved" if there are unsaved shift changes (though browsers no longer let you control the exact warning message.)

**Comment by tonyzeoli** [comment]
@majick777 okay, that sounds reasonable - i was just thinking out loud as to what we do if the user hasn't clicked Update on the page to save their work. But yes, we can just say "Wait, if you navigate away from this page, your changes will not be saved. (Cancel / Save). That's fine. I just wasn't sure, but that does the trick.

**Comment by tonyzeoli** [comment]
@majick777 We may also want a "legend" of sorts to show people what green and yellow mean. If green, your Yellow, your change is NEW but not yet saved. If Green, your change is NEW, but has been saved.

**Comment by majick777** [comment]
@tonyzeoli I've added the background colors to the shifts and a navigate away from page dialogue on shift changes. Also, any changed fields are highlighted with the original bolded. Plus I fixed a bug where newly added shifts were not removable since last few development versions. Check it out, I will leave this open for now in case there are further on this front, I think it's pretty good now, but this was a small enough issue that I decided we could include it in 2.3.3 without it holding that up.
